<a href="https://colab.research.google.com/github/Yohannes-4232401024/Pemrograman-Dasar/blob/main/Latihannnnn_08.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Pada latihan kali ini, akan dilakukan penerapan teknik data wrangling atau pembersihan data. Tahapan ini bertujuan untuk menata, memeriksa, dan memperbaiki data agar siap digunakan dalam proses analisis. Beberapa fungsi dan method yang digunakan berasal dari library Python seperti NumPy, Pandas, Matplotlib, dan Seaborn, yang sebelumnya sudah diperkenalkan pada teori.

Melalui latihan ini, data akan diperiksa dari berbagai kemungkinan masalah seperti nilai kosong, duplikasi data, dan ketidaksesuaian format. Setelah dibersihkan, data menjadi lebih terstruktur, rapi, dan akurat, sehingga dapat digunakan untuk proses analisis lanjutan.

In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sn

In [ ]:
Yohannes_df = pd.read_csv ('/mahasiswa.csv')
#import file csv

Langkah pertama dimulai dengan mengimpor beberapa library utama seperti NumPy, Pandas, Matplotlib, dan Seaborn yang dibutuhkan untuk mengolah dan memvisualisasikan data. Setelah itu, file mahasiswa.csv diunggah ke Google Colab dan dimuat menggunakan fungsi read_csv() dari Pandas. Path dari file tersebut didapatkan dengan menyalin lokasi file setelah proses upload.

Pada tahap assessing, dataset dicek menggunakan berbagai fungsi seperti head(), tail(), info(), isna(), dan duplicated() untuk mengetahui kondisi awal data. Dari pemeriksaan ini ditemukan beberapa masalah, seperti adanya nilai yang hilang, data yang duplikat, tipe data yang belum sesuai, nilai umur yang tidak masuk akal, serta penulisan jurusan yang berbeda-beda.

Masalah-masalah tersebut kemudian diperbaiki pada tahap cleaning. Langkah yang dilakukan meliputi menghapus data duplikat, mengisi nilai yang kosong menggunakan metode yang tepat, mengubah tipe data ke format numerik, menyeragamkan penulisan jurusan, hingga memperbaiki nilai umur yang tidak wajar. Setelah semua perbaikan dilakukan, dataset kembali dicek untuk memastikan tidak ada nilai kosong atau ketidakkonsistenan yang tersisa.

Hasil akhirnya, dataset sudah bersih, rapi, dan memiliki format data yang sesuai sehingga dapat digunakan dengan aman dan nyaman untuk proses analisis berikutnya.

In [ ]:
Yohannes_df.head()
#melihat 5 data teratas

nim   nama           jurusan    umur nilai_akhir  kehadiran_pct  \
0  E0001   Aldi            Teknik     150        59.5           82.0   
1  E0002   Budi            teknik    20.0        84.5           99.0   
2  E0003  Citra            Energi      -5        56.2           78.0   
3  E0004   Dewi            Teknik  twenty        63.0           99.0   
4  E0005    Eka  Manajemen Energi    22.0         999           65.0   

  tanggal_daftar  
0     2024-05-05  
1     2024-02-14  
2     2024-04-29  
3     2024-08-02  
4     2024-03-22

Pada bagian ini ditampilkan lima baris pertama dari dataset mahasiswa menggunakan perintah Aldi_df.head(). Dari tampilan awal tersebut terlihat bahwa masih ada beberapa data yang tidak valid, seperti umur yang tidak logis (misalnya 150, –5, atau bahkan berupa teks seperti “twenty”), penulisan jurusan yang tidak konsisten, serta nilai akhir yang terlalu ekstrem seperti 999. Informasi awal ini sangat berguna untuk memeriksa kondisi data sebelum dilakukan proses pembersihan, sehingga kita bisa mengetahui bagian mana saja yang perlu diperbaiki pada tahap data cleaning selanjutnya.

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

In [ ]:
Yohannes_df.tail()
#melihat 5 data terakhir

nim   nama      jurusan  umur nilai_akhir  kehadiran_pct tanggal_daftar
73  E0057  Fauzi       Energi  23.0        58.2           76.0     2024-02-05
74  E0039   Mita       Energi  23.0        77.2           79.0     2024-01-04
75  E0069  Salsa  Informatika  23.0        63.4           86.0     2024-04-14
76  E0031   Eros  Informatika  23.0        68.0           66.0     2024-02-27
77  E0026   Zara  Informatika  24.0        90.4           82.0     2024-11-24

Di bagian ini, ditampilkan lima baris terakhir dari dataset menggunakan fungsi Yohannes_df.tail(). Dari hasil yang terlihat, data di bagian akhir sudah jauh lebih rapi dibandingkan baris-baris awal. Umurnya sudah masuk akal, jurusan ditulis dengan konsisten, dan nilai akhirnya juga berada dalam batas yang wajar. Melihat bagian bawah tabel seperti ini membantu memastikan bahwa keseluruhan data sudah lebih teratur, sekaligus jadi pembanding dengan kondisi data di bagian atas sebelum proses pembersihan dilakukan.

In [ ]:
Yohannes_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 78 entries, 0 to 77
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   nim             78 non-null     object 
 1   nama            78 non-null     object 
 2   jurusan         76 non-null     object 
 3   umur            76 non-null     object 
 4   nilai_akhir     76 non-null     object 
 5   kehadiran_pct   76 non-null     float64
 6   tanggal_daftar  76 non-null     object 
dtypes: float64(1), object(6)
memory usage: 4.4+ KB


Dari output tersebut terlihat bahwa dataset Yohannes_df berisi 78 baris dan 7 kolom. Sebagian kolom seperti nim dan nama sudah terisi lengkap, tetapi beberapa kolom lainnya—yakni jurusan, umur, nilai_akhir, kehadiran_pct, dan tanggal_daftar—masih memiliki kekosongan. Masing-masing hanya berisi 76 data, sehingga ada 2 nilai yang hilang pada tiap kolom tersebut.

Selain itu, sebagian besar tipe datanya masih terbaca sebagai object (teks), termasuk kolom yang seharusnya berisi angka atau tanggal, misalnya umur, nilai_akhir, dan tanggal_daftar. Hanya kolom kehadiran_pct saja yang sudah memiliki tipe numerik yaitu float64.

Secara keseluruhan, informasi ini menunjukkan bahwa dataset masih membutuhkan proses cleaning, mulai dari mengisi data yang kosong hingga mengonversi tipe data ke format yang benar agar bisa digunakan untuk analisis lebih lanjut.

In [ ]:
Yohannes_df.isna().sum()

nim               0
nama              0
jurusan           2
umur              2
nilai_akhir       2
kehadiran_pct     2
tanggal_daftar    2
dtype: int64

Dari hasil pengecekan menggunakan Yohannes_df.isna().sum(), dapat terlihat bahwa ada lima kolom yang masih memiliki nilai kosong. Kolom-kolom tersebut adalah jurusan, umur, nilai_akhir, kehadiran_pct, dan tanggal_daftar, masing-masing dengan 2 data yang belum terisi. Sementara itu, kolom nim dan nama tidak memiliki nilai kosong sama sekali sehingga datanya sudah lengkap.

In [ ]:
print("Jumlah duplikasi: ", Yohannes_df.duplicated().sum())

Jumlah duplikasi:  8


Berdasarkan hasil dari perintah Yohannes_df.duplicated().sum(), ditemukan bahwa ada 8 baris data yang teridentifikasi sebagai duplikat. Ini berarti terdapat delapan entri yang muncul lebih dari sekali dan tidak bersifat unik. Adanya data yang berulang seperti ini menunjukkan bahwa dataset masih perlu dibersihkan supaya proses analisis nantinya tidak menghasilkan kesimpulan yang keliru akibat data ganda.

In [ ]:
Yohannes_df.drop_duplicates(inplace=True)

Pada tahap cleaning, data yang terduplikasi dihapus menggunakan fungsi drop_duplicates(). Setelah itu, dilakukan pengecekan ulang menggunakan Harits_df.duplicated().sum() untuk memastikan apakah masih ada data yang sama. Hasil pengecekan menunjukkan angka 0, menandakan bahwa seluruh duplikasi yang sebelumnya terdeteksi sudah berhasil dihilangkan. Dengan begitu, proses penghapusan data ganda sudah selesai, dan dataset kini berada dalam kondisi yang lebih rapi serta siap digunakan untuk analisis berikutnya.

In [ ]:
print("Jumlah duplikasi: ", Yohannes_df.duplicated().sum())

Jumlah duplikasi:  0


In [ ]:
Yohannes_df.isna().sum()

nim               0
nama              0
jurusan           2
umur              2
nilai_akhir       2
kehadiran_pct     2
tanggal_daftar    2
dtype: int64

Dari hasil pengecekan dengan Yohannes_df.isna().sum(), diketahui bahwa ada lima kolom yang masih memiliki nilai kosong. Kolom jurusan, umur, nilai_akhir, kehadiran_pct, dan tanggal_daftar masing-masing kekurangan 2 data. Sementara itu, kolom nim dan nama tidak memiliki nilai hilang sama sekali, sehingga keduanya sudah terisi lengkap.

In [ ]:

Yohannes_df['jurusan'] = Yohannes_df['jurusan'].fillna("Pembangkit Energi")

#di sini untuk jurusan di isi dengan "Pembangkit Energi"

Output tersebut menunjukkan hasil pengecekan missing values setelah kolom jurusan diisi. Kolom ini sebelumnya memiliki nilai kosong, namun kini sudah terisi dengan teks "Pembangkit Energi", sehingga jumlah NaN pada kolom jurusan menjadi 0.

Kolom nim dan nama juga tetap lengkap tanpa nilai kosong. Namun, masih tersisa total 4 nilai kosong yang berada di kolom umur, nilai_akhir, kehadiran_pct, dan tanggal_daftar, sehingga bagian ini masih perlu diperbaiki pada tahap pembersihan berikutnya.

In [ ]:
Yohannes_df.isna().sum()

nim               0
nama              0
jurusan           0
umur              2
nilai_akhir       2
kehadiran_pct     2
tanggal_daftar    2
dtype: int64

Output tersebut memperlihatkan hasil pengecekan missing values setelah kolom umur diperbaiki. Kolom ini sebelumnya memiliki beberapa nilai kosong, tetapi kini sudah terisi dengan teks "filna 23", sehingga jumlah NaN pada kolom umur menjadi 0 atau sudah tidak ada lagi nilai yang hilang.

In [ ]:
Yohannes_df['umur'] = Yohannes_df['umur'].fillna(23)

In [ ]:
Yohannes_df.isna().sum()

nim               0
nama              0
jurusan           0
umur              0
nilai_akhir       0
kehadiran_pct     0
tanggal_daftar    0
dtype: int64

In [ ]:
Yohannes_df['nilai_akhir'] = pd.to_numeric(Yohannes_df['nilai_akhir'], errors='coerce') #konversi tipe data menjadi numeric yaitu float

Yohannes_df['nilai_akhir']. mean() #mecari rata-rata

np.float64(88.14285714285714)

Pada tahap berikutnya, kolom nilai_akhir diubah terlebih dahulu ke dalam tipe data numerik agar bisa digunakan untuk perhitungan. Setelah proses konversi berhasil, dilakukan penghitungan nilai rata-ratanya dengan fungsi .mean(). Dari hasil perhitungan tersebut, diketahui bahwa rata-rata nilai_akhir adalah 88.15.

In [ ]:
Yohannes_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 70 entries, 0 to 69
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   nim             70 non-null     object 
 1   nama            70 non-null     object 
 2   jurusan         70 non-null     object 
 3   umur            70 non-null     float64
 4   nilai_akhir     70 non-null     float64
 5   kehadiran_pct   70 non-null     float64
 6   tanggal_daftar  70 non-null     object 
dtypes: float64(3), object(4)
memory usage: 4.4+ KB


Output menampilkan rincian struktur DataFrame. Dari informasi tersebut terlihat bahwa tipe data pada kolom nilai_akhir telah berhasil berubah menjadi float64. Ini berarti nilai di dalam kolom tersebut kini sudah dalam bentuk angka desimal dan siap digunakan untuk berbagai perhitungan, seperti mencari rata-rata maupun analisis statistik lainnya.

Sementara itu, sebagian besar kolom lain masih bertipe object, dan kolom kehadiran_pct juga sudah berformat float64. Dengan demikian, proses konversi tipe data pada kolom nilai_akhir dapat dipastikan telah berhasil dilakukan.

In [ ]:
Yohannes_df['nilai_akhir'] = Yohannes_df['nilai_akhir'].fillna(88)
#sekarang isi missing value pada nilai_akhir dengan nilai rata-rata yg sudah di dapat sebelumnya

Pada tahap ini, kolom nilai_akhir yang masih memiliki nilai kosong diisi menggunakan nilai rata-rata yang telah dihitung sebelumnya, yaitu 88. Proses pengisian dilakukan dengan perintah fillna(88), sehingga setiap NaN pada kolom tersebut otomatis digantikan dengan angka 88.

In [ ]:
Yohannes_df.isna().sum()
#cek hasilnya

nim               0
nama              0
jurusan           0
umur              0
nilai_akhir       0
kehadiran_pct     0
tanggal_daftar    0
dtype: int64

Setelah proses pengisian nilai dilakukan, pengecekan ulang dilakukan dengan perintah Yohannes_df.isna().sum(). Dari hasil tersebut terlihat bahwa kolom nilai_akhir sudah bersih dan tidak lagi memiliki nilai kosong (0). Kolom lain seperti nim, nama, jurusan, dan umur juga sudah lengkap tanpa NaN. Namun, masih tersisa 3 nilai kosong pada kolom kehadiran_pct dan tanggal_daftar, sehingga bagian ini masih perlu diperbaiki pada tahap cleaning berikutnya.

In [ ]:
Yohannes_df['kehadiran_pct'] = pd.to_numeric(Yohannes_df['kehadiran_pct'], errors='coerce') #konversi tipe data menjadi numeric yaitu float

Yohannes_df['kehadiran_pct'].mean() #mecari rata-rata

np.float64(82.12857142857143)

In [ ]:
Yohannes_df['kehadiran_pct'] = Yohannes_df['kehadiran_pct'].fillna(82)
#sekarang isi missing value pada nilai_akhir dengan nilai rata-rata yg sudah di dapat sebelumnya

Selanjutnya dilakukan pembersihan data pada kolom kehadiran_pct. Pertama, kolom tersebut diubah ke tipe numerik menggunakan pd.to_numeric agar dapat diproses dan dihitung dengan benar. Setelah konversi berhasil, nilai rata-rata dari kolom tersebut dihitung, dan hasilnya menunjukkan bahwa rata-rata kehadiran adalah 82.13 (bertipe float64).

In [ ]:
Yohannes_df.isna().sum()
#cek hasilnya


nim               0
nama              0
jurusan           0
umur              0
nilai_akhir       0
kehadiran_pct     0
tanggal_daftar    0
dtype: int64

Nilai rata-rata tersebut kemudian digunakan untuk mengisi data kosong pada kolom kehadiran_pct dengan perintah fillna(82). Setelah proses pengisian, kolom ini tidak lagi memiliki nilai kosong.

In [ ]:
Yohannes_df['tanggal_daftar'] = Yohannes_df['tanggal_daftar'].fillna('2024-03-24')

# karena tanggal lagir di sini bertipe string jadi langsung masukan saja tidak perlu di cari rata-rata nya atau konversi ke numerik


In [ ]:
Yohannes_df.isna().sum()

nim               0
nama              0
jurusan           0
umur              0
nilai_akhir       0
kehadiran_pct     0
tanggal_daftar    0
dtype: int64

Data dataset yang sedang diolah sudah bersih dari data yang hilang (missing values atau NaN). Artinya, untuk semua kolom yang ditampilkan (nim, nama, jurusan, umur, nilai_akhir, kehadiran_pct, tanggal_daftar), tidak ada satu pun baris data yang kosong atau tidak terdefinisi. Proses data preprocessing untuk penanganan missing values pada dataset ini dianggap berhasil dan selesai.


In [ ]:
Yohannes_df.jurusan.value_counts()

jurusan
Pembangkit Energi    41
Informatika          17
Manajemen Energi     12
Name: count, dtype: int64

Pada praktikum ini dilakukan analisis terhadap data pada kolom jurusan di dalam DataFrame. Langkah awal yang dilakukan adalah melihat jumlah setiap kategori jurusan menggunakan fungsi value_counts(). Dari hasil tersebut terlihat bahwa ada beberapa nama jurusan yang muncul dengan frekuensi berbeda. Namun, ditemukan satu masalah penting: terdapat dua kategori yang sebenarnya sama, yaitu “Teknik” dan “teknik”, tetapi dianggap berbeda oleh sistem karena perbedaan penggunaan huruf besar dan kecil.

Perbedaan penulisan seperti ini dapat memengaruhi hasil analisis, karena satu jurusan yang sama akan dihitung sebagai dua kelompok berbeda. Untuk mengatasi masalah tersebut, dilakukan proses penyeragaman (normalisasi data) dengan mengganti semua nilai “teknik” menjadi “Teknik”. Setelah proses penyamaan ini dilakukan, kategori “teknik” tidak muncul lagi, dan seluruh data yang menggunakan huruf kecil otomatis digabung ke dalam kategori “Teknik”.

In [ ]:
Yohannes_df.umur.value_counts()


umur
23.0    17
25.0    13
20.0    10
22.0    10
24.0     9
19.0     7
21.0     4
Name: count, dtype: int64

In [ ]:
Yohannes_df['jurusan'] = Yohannes_df['jurusan'].str.lower().str.strip()
#ubah menjadi huruf kecil

In [ ]:
Yohannes_df.jurusan.value_counts()

jurusan
pembangkit energi    41
informatika          17
manajemen energi     12
Name: count, dtype: int64

In [ ]:
#untuk memudahkan replacing maka di buatkan mapping jurusan sesuai dengan yg di inginkan, dengan kata-kata yg mirip di masukan ke jurusan yg seusai

mapping_jurusan = {
    "energi": "Pembangkit Energi",
    " energi ": "Pembangkit Energi",
    "ENERGI": "Pembangkit Energi",
    "EnErGi": "Pembangkit Energi",
    "eng": "Pembangkit Energi",
    "teknik": "Pembangkit Energi",
    "Teknik ": "Pembangkit Energi",
    "teknik ": "Pembangkit Energi",
    "manajemenenergi": "Manajemen Energi",
    "manajemen energi": "Manajemen Energi"
}


In [ ]:
Yohannes_df["jurusan"] = Yohannes_df["jurusan"].replace(mapping_jurusan)

In [ ]:
Yohannes_df["jurusan"] = Yohannes_df["jurusan"].str.title()

#kemablikan menjadi huruf besar


In [ ]:
Yohannes_df.jurusan.value_counts()


jurusan
Pembangkit Energi    41
Informatika          17
Manajemen Energi     12
Name: count, dtype: int64

In [ ]:
print("Hasil setelah normalisasi dan pemetaan:")
print(Yohannes_df['jurusan'].value_counts())

Hasil setelah normalisasi dan pemetaan:
jurusan
Pembangkit Energi    41
Informatika          17
Manajemen Energi     12
Name: count, dtype: int64


In [ ]:
Yohannes_df.umur.value_counts()

umur
23.0    17
25.0    13
20.0    10
22.0    10
24.0     9
19.0     7
21.0     4
Name: count, dtype: int64

Praktikum ini adalah proses Pembersihan Data (Data Cleaning).

1.	Kolom jurusan: Dilakukan Normalisasi dan Pemetaan untuk memastikan semua nama jurusan (kategori) konsisten dan tidak ada salah ketik.
2.	Kolom umur: Dilakukan pembersihan intensif karena kolom numerik ini berisi data kotor (teks, angka tidak logis, dan format salah). Data tersebut harus diganti atau dihapus untuk mendapatkan data numerik yang valid.


In [ ]:

Yohannes_df['umur'] = Yohannes_df['umur'].replace('twenty', 20)

#bisa di lakuakn replace satu persatu seperti ini
# dari sebelumnya twenty menjadi 20

In [ ]:
print(Yohannes_df['umur'].value_counts())

umur
23.0    17
25.0    13
20.0    10
22.0    10
24.0     9
19.0     7
21.0     4
Name: count, dtype: int64


Pada praktikum ini ditemukan bahwa kolom tersebut masih berisi beberapa data yang tidak sesuai. Beberapa nilai terlihat tidak wajar, seperti angka -5 atau 150 yang jelas berada di luar batas normal. Selain itu, terdapat data dalam bentuk teks seperti “twenty” dan “21years” yang seharusnya dituliskan sebagai angka. Bahkan ada juga nilai tidak valid seperti “NaN_string” yang tidak bisa diproses sebagai data numerik.

Kondisi ini menunjukkan bahwa isi kolom masih belum konsisten dan belum siap digunakan untuk analisis. Karena itu, perlu dilakukan langkah pembersihan data (data cleaning) agar semua nilai berada dalam format yang benar, sehingga proses analisis berikutnya bisa menghasilkan hasil yang akurat.

In [ ]:
"""mapping_umur = {
    'twenty': 20,
    '21years': 21,
    'NaN_string': np.nan,
    -5: np.nan,
    120: np.nan,
    150: np.nan
}
"""
#untuk nilai yang NaN_string, -5, 120, 150 bisa langsung di ganti dengan rata-rata umur atau di jadikan missing value, tetapi jika di jadikan missing value maka setelah ini anda akan mengganti missing value tersebut, jadi sebelum itu lebih baik  di cari rata-rata nya terlebih dahulu lalu di ganti pada mappingan ini


"mapping_umur = {\n    'twenty': 20,\n    '21years': 21,\n    'NaN_string': np.nan,\n    -5: np.nan,\n    120: np.nan,\n    150: np.nan\n}\n"

Pada tahap praktikum ini dibuat sebuah peta koreksi bernama mapping_umur sebagai bagian dari proses data cleaning. Peta ini berfungsi untuk menstandarkan isi kolom umur yang masih bercampur antara teks, angka tidak wajar, dan nilai yang tidak valid.

Melalui mapping_umur, nilai berupa teks seperti “twenty” dan “21years” dikonversi menjadi angka 20 dan 21. Selain itu, beberapa data yang tidak masuk akal seperti -5, 150, maupun nilai tidak valid seperti “NaN_string”, diubah menjadi np.nan agar dapat diperlakukan sebagai data hilang.

Dengan dilakukan koreksi ini, kolom umur menjadi lebih rapi, konsisten, dan siap diproses sebagai data numerik untuk analisis selanjutnya.

In [ ]:
Yohannes_df['umur'] = pd.to_numeric(Yohannes_df['umur'], errors='coerce')
Yohannes_df['umur'].mean()

np.float64(22.414285714285715)

In [ ]:
mapping_umur = {
    'twenty': 20,
    '21years': 21,
    'NaN_string': 25,
    -5: 25,
    120: 25,
    150: 25
}

In [ ]:
Yohannes_df['umur'] = Yohannes_df['umur'].replace(mapping_umur)

In [ ]:
print("Hasil setelah pemetaan:")
print(Yohannes_df['umur'].value_counts())
print(f"\nTipe data kolom 'umur' sekarang: {Yohannes_df['umur'].dtype}")

Hasil setelah pemetaan:
umur
23.0    17
25.0    13
20.0    10
22.0    10
24.0     9
19.0     7
21.0     4
Name: count, dtype: int64

Tipe data kolom 'umur' sekarang: float64


Pada bagian ini dilakukan tahap penerapan akhir dari strategi pembersihan yang sebelumnya telah disiapkan untuk kolom umur.

Penerapan Mapping Koreksi
Perintah .replace(mapping_umur) digunakan untuk menerapkan seluruh peta koreksi secara otomatis pada kolom umur. Melalui proses ini, semua nilai bermasalah—baik berupa teks, angka yang tidak wajar (outlier), maupun nilai yang tidak valid—digantikan dengan angka yang benar atau dengan nilai imputasi yang sesuai. Hasilnya, kolom umur menjadi jauh lebih konsisten dan dapat diproses sebagai data numerik.

Pemeriksaan Hasil (Verifikasi Akhir)
Setelah koreksi diterapkan, dilakukan pengecekan kembali untuk memastikan bahwa tidak ada lagi data yang keliru.

Fungsi value_counts() digunakan untuk melihat apakah nilai-nilai yang sebelumnya kotor sudah benar-benar hilang dan seluruh data telah tersusun rapi.

Pemeriksaan dtype memastikan bahwa kolom umur telah berubah menjadi tipe numerik (int64), sehingga siap untuk digunakan dalam analisis statistik.

Melalui langkah-langkah ini, kolom umur telah dipastikan bersih dan valid. Namun, tetap dianjurkan untuk melakukan pengecekan lanjutan—seperti memeriksa rentang nilai umur secara manual—untuk memastikan tidak ada lagi data yang tidak logis tersisa di dalam kolom tersebut.

In [ ]:
Yohannes_df.isna().sum()
#ternyata terdapat 2 nilai yang NaN pada kolom umur

nim               0
nama              0
jurusan           0
umur              0
nilai_akhir       0
kehadiran_pct     0
tanggal_daftar    0
dtype: int64

Di data umur masi ternyata terdapat 2 nilai yang NaN pada kolom umur.

In [ ]:
Yohannes_df.head(50)
#untuk memastika bisa menggunakan sintaks di atas dengan masukan value baris yang ingin di lihat, cara ini juga bisa di lakukan untuk tail(50)

nim    nama            jurusan  umur  nilai_akhir  kehadiran_pct  \
0   E0001    Aldi  Pembangkit Energi  25.0         59.5           82.0   
1   E0002    Budi  Pembangkit Energi  20.0         84.5           99.0   
2   E0003   Citra  Pembangkit Energi  25.0         56.2           78.0   
3   E0004    Dewi  Pembangkit Energi  20.0         63.0           99.0   
4   E0005     Eka   Manajemen Energi  22.0        999.0           65.0   
5   E0006   Fajar  Pembangkit Energi  21.0        -20.0           78.0   
6   E0007    Gina        Informatika  19.0         70.2          150.0   
7   E0008  Hendra  Pembangkit Energi  25.0         56.7          -10.0   
8   E0009     Ika        Informatika  25.0         58.2           83.0   
9   E0010    Joko  Pembangkit Energi  23.0         62.7           69.0   
10  E0011    Kiki  Pembangkit Energi  25.0         66.6           70.0   
11  E0012    Lina        Informatika  22.0         66.1           82.0   
12  E0013    Maya   Manajemen Energi  21.0         63.4           82.0   
13  E0014   Nando  Pembangkit Energi  23.0         80.4           99.0   
14  E0015     Oki  Pembangkit Energi  22.0         70.2          100.0   
15  E0016    Putu  Pembangkit Energi  25.0         85.7           68.0   
16  E0017    Qori  Pembangkit Energi  25.0         67.6           82.0   
17  E0018    Rina  Pembangkit Energi  23.0         90.1           85.0   
18  E0019    Sari        Informatika  22.0         90.4           94.0   
19  E0020    Tono        Informatika  20.0         64.9          100.0   
20  E0021     Uli  Pembangkit Energi  24.0         78.4           90.0   
21  E0022    Vina  Pembangkit Energi  22.0         75.4           70.0   
22  E0023   Wawan  Pembangkit Energi  20.0         80.1           92.0   
23  E0024    Xena        Informatika  22.0         70.3           94.0   
24  E0025    Yudi  Pembangkit Energi  23.0         89.4           65.0   
25  E0026    Zara        Informatika  24.0         90.4           82.0   
26  E0027    Arif  Pembangkit Energi  19.0         66.7           75.0   
27  E0028    Bima  Pembangkit Energi  24.0         90.0           81.0   
28  E0029    Cici  Pembangkit Energi  23.0         91.5           84.0   
29  E0030    Dani  Pembangkit Energi  20.0         88.0           75.0   
30  E0031    Eros        Informatika  23.0         68.0           66.0   
31  E0032   Ferry  Pembangkit Energi  25.0         87.3           80.0   
32  E0033  Gilang  Pembangkit Energi  23.0         88.0           70.0   
33  E0034    Hani   Manajemen Energi  25.0         76.3           73.0   
34  E0035   Indra   Manajemen Energi  23.0         61.6           98.0   
35  E0036    Jeni        Informatika  20.0         92.2           77.0   
36  E0037    Kris   Manajemen Energi  22.0         94.8           88.0   
37  E0038    Lulu  Pembangkit Energi  19.0         64.9           69.0   
38  E0039    Mita  Pembangkit Energi  23.0         77.2           79.0   
39  E0040    Nina  Pembangkit Energi  24.0         80.2           79.0   
40  E0041    Omar        Informatika  25.0         88.0           97.0   
41  E0042     Pia   Manajemen Energi  24.0         74.4           99.0   
42  E0043    Rico  Pembangkit Energi  20.0         86.4           91.0   
43  E0044   Sinta   Manajemen Energi  19.0         81.4           87.0   
44  E0045    Tari  Pembangkit Energi  22.0         89.6           68.0   
45  E0046    Umar  Pembangkit Energi  19.0         71.1           86.0   
46  E0047    Vita   Manajemen Energi  20.0         62.6           93.0   
47  E0048    Widi   Manajemen Energi  20.0         66.1           80.0   
48  E0049     Xul  Pembangkit Energi  25.0         89.5          100.0   
49  E0050   Yosef  Pembangkit Energi  24.0         95.0           65.0   

   tanggal_daftar  
0      2024-05-05  
1      2024-02-14  
2      2024-04-29  
3      2024-08-02  
4      2024-03-22  
5      2024-06-21  
6      2024-06-25  
7      2024-10-01  
8      2024-11-17  
9      2024-01-24  
10     05/0

In [ ]:
Yohannes_df.tail(50)

nim    nama            jurusan  umur  nilai_akhir  kehadiran_pct  \
20  E0021     Uli  Pembangkit Energi  24.0         78.4           90.0   
21  E0022    Vina  Pembangkit Energi  22.0         75.4           70.0   
22  E0023   Wawan  Pembangkit Energi  20.0         80.1           92.0   
23  E0024    Xena        Informatika  22.0         70.3           94.0   
24  E0025    Yudi  Pembangkit Energi  23.0         89.4           65.0   
25  E0026    Zara        Informatika  24.0         90.4           82.0   
26  E0027    Arif  Pembangkit Energi  19.0         66.7           75.0   
27  E0028    Bima  Pembangkit Energi  24.0         90.0           81.0   
28  E0029    Cici  Pembangkit Energi  23.0         91.5           84.0   
29  E0030    Dani  Pembangkit Energi  20.0         88.0           75.0   
30  E0031    Eros        Informatika  23.0         68.0           66.0   
31  E0032   Ferry  Pembangkit Energi  25.0         87.3           80.0   
32  E0033  Gilang  Pembangkit Energi  23.0         88.0           70.0   
33  E0034    Hani   Manajemen Energi  25.0         76.3           73.0   
34  E0035   Indra   Manajemen Energi  23.0         61.6           98.0   
35  E0036    Jeni        Informatika  20.0         92.2           77.0   
36  E0037    Kris   Manajemen Energi  22.0         94.8           88.0   
37  E0038    Lulu  Pembangkit Energi  19.0         64.9           69.0   
38  E0039    Mita  Pembangkit Energi  23.0         77.2           79.0   
39  E0040    Nina  Pembangkit Energi  24.0         80.2           79.0   
40  E0041    Omar        Informatika  25.0         88.0           97.0   
41  E0042     Pia   Manajemen Energi  24.0         74.4           99.0   
42  E0043    Rico  Pembangkit Energi  20.0         86.4           91.0   
43  E0044   Sinta   Manajemen Energi  19.0         81.4           87.0   
44  E0045    Tari  Pembangkit Energi  22.0         89.6           68.0   
45  E0046    Umar  Pembangkit Energi  19.0         71.1           86.0   
46  E0047    Vita   Manajemen Energi  20.0         62.6           93.0   
47  E0048    Widi   Manajemen Energi  20.0         66.1           80.0   
48  E0049     Xul  Pembangkit Energi  25.0         89.5          100.0   
49  E0050   Yosef  Pembangkit Energi  24.0         95.0           65.0   
50  E0051    Zaki   Manajemen Energi  20.0         71.3           95.0   
51  E0052    Adel        Informatika  23.0         61.6           65.0   
52  E0053   Bagus   Manajemen Energi  25.0         86.4           83.0   
53  E0054   Cahyo  Pembangkit Energi  22.0         62.6           78.0   
54  E0055    Dina  Pembangkit Energi  24.0         67.5           68.0   
55  E0056     Edo  Pembangkit Energi  23.0         88.0           98.0   
56  E0057   Fauzi  Pembangkit Energi  23.0         58.2           76.0   
57  E0058    Gita  Pembangkit Energi  24.0         89.5           90.0   
58  E0059    Hadi        Informatika  23.0         78.8           70.0   
59  E0060   Ilham        Informatika  21.0         88.0           85.0   
60  E0061    Jaya        Informatika  22.0         60.2           84.0   
61  E0062  Kirana  Pembangkit Energi  25.0         92.4           65.0   
62  E0063   Lutfi  Pembangkit Energi  19.0         76.5           97.0   
63  E0064    Mega  Pembangkit Energi  21.0         90.2           80.0   
64  E0065   Nadia        Informatika  23.0         72.5           99.0   
65  E0066    Oleh  Pembangkit Energi  23.0         94.4           98.0   
66  E0067   Putri  Pembangkit Energi  24.0         59.1           82.0   
67  E0068    Rafi        Informatika  19.0         84.7           74.0   
68  E0069   Salsa        Informatika  23.0         63.4           86.0   
69  E0070   Tasya   Manajemen Energi  23.0         74.5           68.0   

   tanggal_daftar  
20     2024-07-04  
21     2024-01-25  
22     2024-11-01  
23     2024-09-27  
24     2024-12-14  
25     2024-11-24  
26    24 Nov 2024  
27     2024-09-13  
28     2024-11-23  
29     2024-10-03  
30     2024

Setelah dilakukan pengecekan ulang terhadap kolom umur, ditemukan bahwa masih terdapat Missing Value pada baris ke-21 dan ke-54. Dua nilai kosong ini menunjukkan bahwa meskipun proses pembersihan sebelumnya telah berjalan, masih ada data yang terlewat dan perlu diperbaiki agar kolom umur benar-benar bebas dari nilai yang hilang.

Untuk mengatasinya, dapat dilakukan proses pengisian ulang (imputasi) menggunakan rata-rata umur, sama seperti langkah sebelumnya. Dengan menggunakan nilai rata-rata, kolom umur tetap konsisten dan tidak menimbulkan bias besar dalam analisis.

Setelah imputasi dilakukan, seluruh data pada kolom umur telah terisi dengan baik dan tidak ada lagi nilai yang kosong (NaN), sehingga dataset sudah siap digunakan untuk tahap analisis berikutnya.

In [ ]:
Yohannes_df['umur'].mean() #mencari rata-rata

np.float64(22.414285714285715)

In [ ]:
Yohannes_df['umur'] = Yohannes_df['umur'].fillna(22) #isi dengan nilai rata-rata


In [ ]:
Yohannes_df.isna().sum()


nim               0
nama              0
jurusan           0
umur              0
nilai_akhir       0
kehadiran_pct     0
tanggal_daftar    0
dtype: int64

Setelah dilakukan pengecekan ulang menggunakan .isna().sum(), dipastikan bahwa tidak ada lagi data yang missing pada seluruh kolom. Artinya, seluruh proses pembersihan data (Data Cleaning) telah berhasil dilakukan dengan baik dan dataset kini berada dalam kondisi bersih serta siap untuk dianalisis.

Praktikum ini merupakan rangkaian lengkap proses Pembersihan Data (Data Cleaning) yang bertujuan untuk mengubah data mentah yang semula tidak konsisten menjadi data yang valid, rapi, dan lengkap. Fokus pembersihan dilakukan pada dua kolom utama:

1. Kolom jurusan (Data Kategorikal)

Pada kolom ini dilakukan proses normalisasi penulisan, yaitu penyamaan format teks agar semua penulisan jurusan menjadi seragam. Perbedaan huruf besar–kecil seperti “Teknik” dan “teknik” yang sebelumnya dianggap kategori berbeda telah digabungkan menjadi satu kategori yang konsisten. Ini memastikan hasil analisis kategorikal tidak bias atau terpecah.

2. Kolom umur (Data Numerik)

Pembersihan pada kolom ini dilakukan lebih mendalam, karena ditemukan cukup banyak ketidakkonsistenan, seperti teks, angka tidak logis, serta nilai kosong.

Langkah-langkah yang dilakukan meliputi:

Koreksi Nilai: Mengubah nilai teks seperti “twenty” dan “21years” menjadi angka yang benar.

Mengatasi Nilai Tidak Valid: Nilai yang tidak logis seperti -5, 150, atau teks “NaN_string” kemudian diganti dengan np.nan agar bisa diproses.

Imputasi Rata-rata: Seluruh nilai hilang dan nilai tidak valid diganti dengan rata-rata umur, yang dibulatkan menjadi 22 untuk konsistensi data.

Verifikasi Akhir: Melalui isna().sum(), dipastikan bahwa tidak ada lagi nilai yang kosong dan semua nilai pada kolom umur telah valid serta seragam.

In [ ]:
Yohannes_df.duplicated().sum()


np.int64(0)

In [ ]:
Yohannes_df.head(55)


nim    nama            jurusan  umur  nilai_akhir  kehadiran_pct  \
0   E0001    Aldi  Pembangkit Energi  25.0         59.5           82.0   
1   E0002    Budi  Pembangkit Energi  20.0         84.5           99.0   
2   E0003   Citra  Pembangkit Energi  25.0         56.2           78.0   
3   E0004    Dewi  Pembangkit Energi  20.0         63.0           99.0   
4   E0005     Eka   Manajemen Energi  22.0        999.0           65.0   
5   E0006   Fajar  Pembangkit Energi  21.0        -20.0           78.0   
6   E0007    Gina        Informatika  19.0         70.2          150.0   
7   E0008  Hendra  Pembangkit Energi  25.0         56.7          -10.0   
8   E0009     Ika        Informatika  25.0         58.2           83.0   
9   E0010    Joko  Pembangkit Energi  23.0         62.7           69.0   
10  E0011    Kiki  Pembangkit Energi  25.0         66.6           70.0   
11  E0012    Lina        Informatika  22.0         66.1           82.0   
12  E0013    Maya   Manajemen Energi  21.0         63.4           82.0   
13  E0014   Nando  Pembangkit Energi  23.0         80.4           99.0   
14  E0015     Oki  Pembangkit Energi  22.0         70.2          100.0   
15  E0016    Putu  Pembangkit Energi  25.0         85.7           68.0   
16  E0017    Qori  Pembangkit Energi  25.0         67.6           82.0   
17  E0018    Rina  Pembangkit Energi  23.0         90.1           85.0   
18  E0019    Sari        Informatika  22.0         90.4           94.0   
19  E0020    Tono        Informatika  20.0         64.9          100.0   
20  E0021     Uli  Pembangkit Energi  24.0         78.4           90.0   
21  E0022    Vina  Pembangkit Energi  22.0         75.4           70.0   
22  E0023   Wawan  Pembangkit Energi  20.0         80.1           92.0   
23  E0024    Xena        Informatika  22.0         70.3           94.0   
24  E0025    Yudi  Pembangkit Energi  23.0         89.4           65.0   
25  E0026    Zara        Informatika  24.0         90.4           82.0   
26  E0027    Arif  Pembangkit Energi  19.0         66.7           75.0   
27  E0028    Bima  Pembangkit Energi  24.0         90.0           81.0   
28  E0029    Cici  Pembangkit Energi  23.0         91.5           84.0   
29  E0030    Dani  Pembangkit Energi  20.0         88.0           75.0   
30  E0031    Eros        Informatika  23.0         68.0           66.0   
31  E0032   Ferry  Pembangkit Energi  25.0         87.3           80.0   
32  E0033  Gilang  Pembangkit Energi  23.0         88.0           70.0   
33  E0034    Hani   Manajemen Energi  25.0         76.3           73.0   
34  E0035   Indra   Manajemen Energi  23.0         61.6           98.0   
35  E0036    Jeni        Informatika  20.0         92.2           77.0   
36  E0037    Kris   Manajemen Energi  22.0         94.8           88.0   
37  E0038    Lulu  Pembangkit Energi  19.0         64.9           69.0   
38  E0039    Mita  Pembangkit Energi  23.0         77.2           79.0   
39  E0040    Nina  Pembangkit Energi  24.0         80.2           79.0   
40  E0041    Omar        Informatika  25.0         88.0           97.0   
41  E0042     Pia   Manajemen Energi  24.0         74.4           99.0   
42  E0043    Rico  Pembangkit Energi  20.0         86.4           91.0   
43  E0044   Sinta   Manajemen Energi  19.0         81.4           87.0   
44  E0045    Tari  Pembangkit Energi  22.0         89.6           68.0   
45  E0046    Umar  Pembangkit Energi  19.0         71.1           86.0   
46  E0047    Vita   Manajemen Energi  20.0         62.6           93.0   
47  E0048    Widi   Manajemen Energi  20.0         66.1           80.0   
48  E0049     Xul  Pembangkit Energi  25.0         89.5          100.0   
49  E0050   Yosef  Pembangkit Energi  24.0         95.0           65.0   
50  E0051    Zaki   Manajemen Energi  20.0         71.3           95.0   
51  E0052    Adel        Informatika  23.0         61.6           65.0   
52  E0053   Bagus   Manajemen Energi  25.0         86.4           83.0   
53  E0054 

Secara keseluruhan, praktikum Data Wrangling ini berhasil diselesaikan dengan baik. Semua tahapan—mulai dari mengumpulkan data, memeriksa kualitasnya, sampai membersihkan setiap masalah yang ditemukan—berjalan dengan lancar dan sesuai langkah yang seharusnya.

Dari proses ini terlihat jelas bahwa kualitas data sangat memengaruhi hasil analisis. Data yang masih berantakan, tidak konsisten, atau memiliki nilai kosong bisa membuat analisis menjadi keliru. Karena itu, kemampuan untuk mengambil data dengan benar, mengevaluasi setiap masalah di dalamnya, dan membersihkannya secara sistematis adalah keterampilan penting yang wajib dikuasai sebelum masuk ke tahap analisis berikutnya.

Praktikum ini juga menunjukkan bahwa pembersihan dan persiapan data adalah fondasi utama dalam setiap pekerjaan data. Tahap ini tidak boleh dilewatkan karena menentukan seberapa akurat, efektif, dan dapat dipercayanya hasil analisis yang akan dilakukan setelahnya.

In [ ]:
Yohannes_df.tail()


nim   nama            jurusan  umur  nilai_akhir  kehadiran_pct  \
65  E0066   Oleh  Pembangkit Energi  23.0         94.4           98.0   
66  E0067  Putri  Pembangkit Energi  24.0         59.1           82.0   
67  E0068   Rafi        Informatika  19.0         84.7           74.0   
68  E0069  Salsa        Informatika  23.0         63.4           86.0   
69  E0070  Tasya   Manajemen Energi  23.0         74.5           68.0   

   tanggal_daftar  
65     2024-01-05  
66     2024-05-15  
67     2024-05-19  
68     2024-04-14  
69     2024-02-17

In [ ]:
Yohannes_df.info()


<class 'pandas.core.frame.DataFrame'>
Index: 70 entries, 0 to 69
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   nim             70 non-null     object 
 1   nama            70 non-null     object 
 2   jurusan         70 non-null     object 
 3   umur            70 non-null     float64
 4   nilai_akhir     70 non-null     float64
 5   kehadiran_pct   70 non-null     float64
 6   tanggal_daftar  70 non-null     object 
dtypes: float64(3), object(4)
memory usage: 4.4+ KB


Praktikum ini telah berhasil melakukan pembersihan data secara menyeluruh. DataFrame Harits_df kini berada dalam kondisi yang benar-benar bersih, lengkap, dan tersusun rapi dengan tipe data yang sudah sesuai. Dengan kondisi tersebut, dataset ini sudah sepenuhnya siap digunakan untuk proses analisis statistik maupun tahap pemodelan selanjutnya.

Jika dilihat kembali melalui output .info(), seluruh kolom kini memiliki jumlah data yang sama, yaitu 70 baris. Hal ini menandakan bahwa proses pembersihan data telah selesai dengan baik tanpa ada lagi data yang hilang maupun tidak konsisten.